rollout/canary.go

```diff
@@ -360,6 +360,11 @@ func (c *rolloutContext) syncRolloutStatusCanary() error {
 	newStatus.AvailableReplicas = replicasetutil.GetAvailableReplicaCountForReplicaSets(c.allRSs)
 	newStatus.HPAReplicas = replicasetutil.GetActualReplicaCountForReplicaSets(c.allRSs)
 	newStatus.Selector = metav1.FormatLabelSelector(c.rollout.Spec.Selector)
+
+	if c.rollout.Spec.Strategy.Canary != nil && c.rollout.Spec.Strategy.Canary.TrafficRouting != nil && !c.rollout.Spec.Strategy.Canary.DynamicStableScale && c.stableRS != nil {
+		newStatus.Selector = metav1.FormatLabelSelector(c.stableRS.Spec.Selector)
+	}
```

Comment:
Add docs because the code is not self explaining on why this change is necessary.

```suggestion
	if c.rollout.Spec.Strategy.Canary != nil && c.rollout.Spec.Strategy.Canary.TrafficRouting != nil && !c.rollout.Spec.Strategy.Canary.DynamicStableScale && c.stableRS != nil {
		// When using traffic routed canary without scaling down the stable replicaset, the number of pods
		// selected by the rollout selector will be up to twice the amount of desired spec.replicas.
		// We update the selector to select the stable pods since the HPA expects a
		// label that queries over pods that should match the replicas count.
		newStatus.Selector = metav1.FormatLabelSelector(c.stableRS.Spec.Selector)
	}
```